aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -89,18 +100,34 @@ module aptos_token_objects::token {
 
         let collection_addr = collection::create_collection_address(&creator_address, &collection_name);
         let collection = object::address_to_object<Collection>(collection_addr);
-        let id = collection::increment_supply(&collection, signer::address_of(&object_signer));
+
+        let deprecated_index = 0;
+        let index = if (features::concurrent_token_v2_enabled()) {
+            option::destroy_with_default(
+                collection::increment_concurrent_supply(&collection, signer::address_of(&object_signer)),
+                aggregator_v2::create_snapshot<u64>(0)
+            )
+        } else {
+            let created_index = collection::increment_supply(&collection, signer::address_of(&object_signer));
+            deprecated_index = option::destroy_with_default(created_index, 0);
+            aggregator_v2::create_snapshot<u64>(deprecated_index)
+        };
 
         let token = Token {
             collection,
-            index: option::get_with_default(&mut id, 0),
+            deprecated_index,
             description,
             name,
             uri,
             mutation_events: object::new_event_handle(&object_signer),
         };
         move_to(&object_signer, token);
 
+        let token_appendix_1 = TokenAppendix1 {
+            index,
+        };
+        move_to(&object_signer, token_appendix_1);
```

Comment:
So, we are create token_appendix_1 independent of whether `concurrent_token_v2_enabled` feature flag is enabled?